sql/src/main/java/io/crate/netty/CrateChannelBootstrapFactory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.netty;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.transport.netty4.Netty4Transport;
+
+import static org.elasticsearch.common.util.concurrent.EsExecutors.daemonThreadFactory;
+
+/**
+ * Factory utility for creating channel server bootstraps, based on the relevant netty {@link Settings}
+ */
+public final class CrateChannelBootstrapFactory {
+
+    private CrateChannelBootstrapFactory() {
+    }
+
+    public static ServerBootstrap newChannelBootstrap(String id, Settings settings) {
```

Comment:
Wonder if this factory is actually limiting if you want to specify custom options for the channel. `Settings` doesn't really let you parameterize since they are the same for both types of channel. I think it would be better to move this into two static methods in `PostgresNetty` and `Netty4MqttServerTransport`. That gives us easy access to modify pipelines options/configuration for both.

Reply:
But the returned ServerBootstrap is further customized as needed for Postgres/Mqtt. 
This code was copy pasted in both places and I think obeying the netty Settings is something that shouldn't be optional.